# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/wmf/WMFMediaDataDecoder.cpp
# Commit: 9cd1ec508d2c
# Full Hash: 9cd1ec508d2c8f63839832def9d911ac082b9c13
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2024-07-23 09:34:47
# Regressor Bug: 1905022
# File Overlap Count: 1
# Description:
#   Bug 1905022 - Exit WMFMediaDataDecoder::ProcessOutput() if video track uses zero video frame copy and it gets one output r=media-playback-reviewers,alwu
#   
#   There were cases that WMFVideoMFTManager::Output() was called more than 10 times within WMFVideoMFTManager::Output(). In this case, WMFVideoMFTManager::Output() did not exit when zero video frame copy was used.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D216775
# ==============================================================================

diff -r 98dbbfb6afd0 -r 9cd1ec508d2c dom/media/platforms/wmf/WMFMediaDataDecoder.cpp
--- a/dom/media/platforms/wmf/WMFMediaDataDecoder.cpp	Tue Jul 23 01:21:05 2024 +0000
+++ b/dom/media/platforms/wmf/WMFMediaDataDecoder.cpp	Tue Jul 23 03:06:37 2024 +0000
@@ -156,6 +156,11 @@
       mInputTimesSet.clear();
     }
     aResults.AppendElement(std::move(output));
+    // If zero-copy video is enabled, multiple WMFVideoMFTManager::Output()
+    // calls must be prevented within WMFMediaDataDecoder::ProcessOutput().
+    if (mMFTManager->UseZeroCopyVideoFrame()) {
+      break;
+    }
     if (mDrainStatus == DrainStatus::DRAINING) {
       break;
     }